addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -3278,14 +3279,100 @@ export class OdooEditor extends EventTarget {
         } else if (ev.key === 'Tab') {
             // Tab
             const sel = this.document.getSelection();
-            const closestTag = (closestElement(sel.anchorNode, 'li, table') || {}).tagName;
-
-            if (closestTag === 'LI') {
-                this._applyCommand('indentList', ev.shiftKey ? 'outdent' : 'indent');
-            } else if (closestTag === 'TABLE') {
+            if (closestElement(sel.anchorNode, 'table')) {
                 this._onTabulationInTable(ev);
-            } else if (!ev.shiftKey) {
+            } else if (ev.shiftKey) {
+                // Split traversed nodes into list items and the rest.
+                const traversedNodes = getTraversedNodes(this.editable);
+                const nonListItems = [];
+                const listItems = new Set(
+                    traversedNodes.map(node => {
+                        const listItem = closestElement(node, 'li');
+                        if (listItem) {
+                            return listItem;
+                        } else if (!(node.querySelector && node.querySelector('li'))) {
+                            nonListItems.push(node);
+                        }
+                    }).filter(node => node && !node.querySelector('li')));
```

Comment:
Yes it's the same check but I hadn't found an elegant way to combine them. Of course, not using map was the solution :-P Thanks!

Like this rather because the  filter is for both as you pointed out:
```js
const listItems = new Set();
const nonListItems = new Set();
for (const node in getTraversedNodes(this.editable)) {
    const closestLi = closestElement(node, 'li');
    const target = closestLi || node;
    if (!(target.querySelector && target.querySelector('li'))) {
        if (closestLi) {
            listItems.add(closestLi);
        } else {
            nonListItems.add(node);
        }
    }
}
```